src/OpenTelemetry.Api/.publicApi/net452/PublicAPI.Unshipped.txt

```diff
@@ -0,0 +1,327 @@
+OpenTelemetry.Metrics.CounterMetric<T>
+OpenTelemetry.Metrics.CounterMetric<T>.CounterMetric() -> void
+OpenTelemetry.Metrics.DoubleObserverMetric
+OpenTelemetry.Metrics.DoubleObserverMetric.DoubleObserverMetric() -> void
+OpenTelemetry.Metrics.DoubleObserverMetricHandle
+OpenTelemetry.Metrics.DoubleObserverMetricHandle.DoubleObserverMetricHandle() -> void
+OpenTelemetry.Metrics.Int64ObserverMetric
+OpenTelemetry.Metrics.Int64ObserverMetric.Int64ObserverMetric() -> void
+OpenTelemetry.Metrics.Int64ObserverMetricHandle
+OpenTelemetry.Metrics.Int64ObserverMetricHandle.Int64ObserverMetricHandle() -> void
+OpenTelemetry.Metrics.LabelSet
+OpenTelemetry.Metrics.LabelSet.LabelSet() -> void
+OpenTelemetry.Metrics.MeasureMetric<T>
+OpenTelemetry.Metrics.MeasureMetric<T>.MeasureMetric() -> void
+OpenTelemetry.Metrics.MeasureMetric<T>.Record(in OpenTelemetry.Baggage context, T value, OpenTelemetry.Metrics.LabelSet labelset) -> void
+OpenTelemetry.Metrics.MeasureMetric<T>.Record(in OpenTelemetry.Baggage context, T value, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>> labels) -> void
+OpenTelemetry.Metrics.MeasureMetric<T>.Record(in OpenTelemetry.Trace.SpanContext context, T value, OpenTelemetry.Metrics.LabelSet labelset) -> void
+OpenTelemetry.Metrics.MeasureMetric<T>.Record(in OpenTelemetry.Trace.SpanContext context, T value, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>> labels) -> void
+OpenTelemetry.Metrics.Meter
+OpenTelemetry.Metrics.Meter.Meter() -> void
+OpenTelemetry.Metrics.MeterFactoryBase
+OpenTelemetry.Metrics.MeterFactoryBase.MeterFactoryBase() -> void
+OpenTelemetry.Metrics.MeterProvider
+OpenTelemetry.Metrics.MeterProvider.Dispose() -> void
+OpenTelemetry.Metrics.MeterProvider.MeterProvider() -> void
+OpenTelemetry.Metrics.NoopBoundCounterMetric<T>
+OpenTelemetry.Metrics.NoopBoundCounterMetric<T>.NoopBoundCounterMetric() -> void
+OpenTelemetry.Metrics.NoopBoundMeasureMetric<T>
+OpenTelemetry.Metrics.NoopBoundMeasureMetric<T>.NoopBoundMeasureMetric() -> void
+OpenTelemetry.Metrics.NoopCounterMetric<T>
+OpenTelemetry.Metrics.NoopCounterMetric<T>.NoopCounterMetric() -> void
+OpenTelemetry.Metrics.NoopDoubleObserverMetric
+OpenTelemetry.Metrics.NoopDoubleObserverMetric.NoopDoubleObserverMetric() -> void
+OpenTelemetry.Metrics.NoopDoubleObserverMetricHandle
+OpenTelemetry.Metrics.NoopDoubleObserverMetricHandle.NoopDoubleObserverMetricHandle() -> void
+OpenTelemetry.Metrics.NoopInt64ObserverMetric
+OpenTelemetry.Metrics.NoopInt64ObserverMetric.NoopInt64ObserverMetric() -> void
+OpenTelemetry.Metrics.NoopInt64ObserverMetricHandle
+OpenTelemetry.Metrics.NoopInt64ObserverMetricHandle.NoopInt64ObserverMetricHandle() -> void
+OpenTelemetry.Metrics.NoopMeasureMetric<T>
+OpenTelemetry.Metrics.NoopMeasureMetric<T>.NoopMeasureMetric() -> void
+OpenTelemetry.Trace.ActivityExtensions
+OpenTelemetry.Trace.IActivityEnumerator<T>
```

Comment:
How is this being used?

Reply:
below one example:
```csharp
private struct LinkEnumerationState : IActivityEnumerator<ActivityLink>
        {
            public bool Created;

            public PooledList<JaegerSpanRef> SpanRefs;

            public bool ForEach(ActivityLink activityLink)
            {
                if (!this.Created)
                {
                    this.SpanRefs = PooledList<JaegerSpanRef>.Create();
                    this.Created = true;
                }

                PooledList<JaegerSpanRef>.Add(ref this.SpanRefs, activityLink.ToJaegerSpanRef());

                return true;
            }
        }
```

Reply:
we need to review and see if we need to make any of these public.

Reply:
@CodeBlanch what are your thoughts on making this public? These are pure .NET core, nothing specific to Otel. So I think we should not make these public APIs.
We can used sharedfile/internal for use within the repo.

Reply:
`IActivityEnumerator` is used by a few extensions off of Activity (and friends):
```
EnumerateTags (Activity tags, ActivityLink tags, or ActivityEvent tags)
EnumerateLinks
EnumerateEvents 
```
I made those extensions public so that people writing exporters (& perhaps instrumentation) would be able to enumerate over all the different collections branching off of Activity in an allocation-free way. If that's not a compelling enough reason, we can make them internal which will allow the interface to also be made internal.

I was just looking at the PR where we made PropertyFetcher internal ([#1466](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1466)), & @ejsmith's comments on there. This falls into that same category. Some of the tools/tricks we use to make the first-party stuff work great would be nice to share with the community.

Reply:
While making them public surely helps more people, it'll become increasing difficult to manage all the public APIs exposed. They are not a Otel spec based API, and we'll have to keep supporting it for a long time once shipped. Most of them are hacky/tricky way and not straightforward to use without writing docs about it.
(I have been copy/pasting ideas/code from this repo for my own exporters required inside Microsoft)

My personal vote would be to mark all such APIs internal, except the ones required by Otel spec.
Then on a case-by-case basis we can decide if we want to expose some of them public, if there is strong need, and the API usage is clear/well documented.